src/IIIFPresentation/BackgroundHandler/BatchCompletion/BatchCompletionMessageHandler.cs

```diff
@@ -184,11 +189,87 @@ private void UpdateCanvasPaintings(Batch batch, Dictionary<AssetId, Canvas> item
             canvasPainting.Thumbnail = thumbnailPath != null ? new Uri(thumbnailPath) : null;
             canvasPainting.Ingesting = false;
             canvasPainting.Modified = DateTime.UtcNow;
-            canvasPainting.StaticHeight = item.Height;
-            canvasPainting.StaticWidth = item.Width;
+
+            if (canvasPainting is {StaticWidth: null, StaticHeight: null})
+            {
+                // #232: set static width/height if not provided.
+                // if canvas painting comes with statics, use them
+                // otherwise, if we can find dimensions within, use those
+                // ReSharper disable once InvertIf
+                if (GetCanvasDimensions(item) is var (width, height))
+                {
+                    canvasPainting.StaticWidth = width;
+                    canvasPainting.StaticHeight = height;
+                }
+            }
+            else if (canvasPainting is {StaticWidth: { } staticWidth, StaticHeight: { } staticHeight}
+                     && item.Items?.GetFirstPaintingAnnotation()?.Body is Image image
+                     && manifest.SpaceId.HasValue) // required for ImageRequest parsing and modification
+            {
+                // #232: if static_width/height provided in canvasPainting
+                // then don't use ones from NQ
+                // and set the body to those dimensions + resized id (image request uri)
+
+                image.Width = staticWidth;
+                image.Height = staticHeight;
+
+                image.Id = pathGenerator.GetModifiedImageRequest(image.Id, manifest.CustomerId,
+                    manifest.SpaceId.Value, staticWidth, staticHeight);
+            }
         }
     }
 
+    private static (int width, int height)? GetCanvasDimensions(Canvas canvas)
```

Comment:
Worth moving these to a reusable utility function, there is a `ManifestX` somewhere (where `GetFirstPaintingAnnotation()` etc are). Would allow for testing in isolation.